src/libs/TransactionPreviewUtils.ts

```diff
@@ -0,0 +1,326 @@
+
+    const requestMerchant = truncate(merchant, {length: CONST.REQUEST_PREVIEW.MAX_LENGTH});
+    const description = truncate(StringUtils.lineBreaksToSpaces(requestComment), {length: CONST.REQUEST_PREVIEW.MAX_LENGTH});
+
+    const isMoneyRequestSettled = isSettled(iouReport?.reportID);
+    const isApproved = isReportApproved({report: iouReport});
+    const isSettlementOrApprovalPartial = !!iouReport?.pendingFields?.partial;
+
+    const hasViolationsOfTypeNotice = hasNoticeTypeViolation(transaction?.transactionID, violations, true) && iouReport && isPaidGroupPolicy(iouReport);
+    const hasFieldErrors = hasMissingSmartscanFields(transaction);
+
+    const isFetchingWaypoints = isFetchingWaypointsFromServer(transaction);
+
+    const isTransactionOnHold = isOnHold(transaction);
+    const isFullySettled = isMoneyRequestSettled && !isSettlementOrApprovalPartial;
+    const isFullyApproved = isApproved && !isSettlementOrApprovalPartial;
+
+    const shouldDisableOnPress = isBillSplit && isEmptyObject(transaction);
+    const shouldShowSkeleton = isEmptyObject(transaction) && !isMessageDeleted(action) && action.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
+    const shouldShowTag = !!tag && isReportAPolicyExpenseChat;
+    const shouldShowCategory = !!category && isReportAPolicyExpenseChat;
+
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    const hasAnyViolations = hasViolationsOfTypeNotice || hasWarningTypeViolation(transaction?.transactionID, violations, true) || hasViolation(transaction, violations, true);
+    const hasErrorOrOnHold = hasFieldErrors || (!isFullySettled && !isFullyApproved && isTransactionOnHold);
+    const shouldShowRBR = hasAnyViolations || hasErrorOrOnHold;
+
+    // When there are no settled transactions in duplicates, show the "Keep this one" button
+    const shouldShowKeepButton = areThereDuplicates;
+    const shouldShowSplitShare = isBillSplit && !!requestAmount && requestAmount > 0;
+
+    /*
+ Show the merchant for IOUs and expenses only if:
+ - the merchant is not empty, is custom, or is not related to scanning smartscan;
+ - the expense is not a distance expense with a pending route and amount = 0 - in this case,
+   the merchant says: "Route pending...", which is already shown in the amount field;
+*/
+    const shouldShowMerchant =
+        !!requestMerchant &&
+        requestMerchant !== CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT &&
+        requestMerchant !== CONST.TRANSACTION.DEFAULT_MERCHANT &&
+        !(isFetchingWaypoints && !requestAmount);
+    const shouldShowDescription = !!description && !shouldShowMerchant && !isScanning;
+
+    return {
+        shouldDisableOnPress,
+        shouldShowSkeleton,
+        shouldShowTag,
+        shouldShowRBR,
+        shouldShowCategory,
+        shouldShowKeepButton,
+        shouldShowSplitShare,
+        shouldShowMerchant,
+        shouldShowDescription,
+    };
```

Comment:
Would be amazing to add unit tests for all these conditions, could you add it maybe in future PR?